### PR TITLE
LPS-165561 Use application default credentials if service account key is not provide

### PR DESCRIPTION
**Pull request by brandizzi**
# References

* This PR is a follow-up to https://github.com/liferay-core-infra/liferay-portal/pull/2153
* LPS Ticket: [LPS-165561](https://issues.liferay.com/browse/LPS-165561)
* LCD Ticket: [LCD-14380](https://issues.liferay.com/browse/LCD-14380)
* How it was tested: [Tests done](https://github.com/liferay-core-infra/liferay-portal/pull/2005#issuecomment-1278113769) (There are testing instructions in the LPS ticket, too.)

# Background

When no credentials are explicitly set, Google's libraries automatically attempt to federate access control as follows:

* If the environment variable `GOOGLE_APPLICATION_CREDENTIALS` is set, ADC uses the service account key or configuration file that the variable points to.
* If the environment variable `GOOGLE_APPLICATION_CREDENTIALS` isn't set, ADC uses the service account that is attached to the resource that is running your code. (workload identity)

As a best practice, Liferay Cloud is moving to leverage Workload Identity for access control across our services that interact with GCP directly. As we are now looking to move our Document Library to GCSStore for all customers, our Liferay service will now be doing just that.

This change can thus also be used by anyone running Liferay in a GKE and is not strictly limited to our use case in Cloud.

# Important note

This pull request has one more change: we also made `aes256Key` optional. This attribute was already optional, but we were forced to add it as an empty value to all files. We suspect it was not the intended behavior, so I removed it here. If there is some reason to make `aes256Key` mandatory, even if empty, please let us know!

### CODE REVIEW

**Review by dantewang — APPROVED** [review]

**Comment by dantewang** [comment]
@zhangyan0701 CI seems to be stuck for this PR, could you take a look?

**Comment by dantewang** [comment]
ci:stop

**Comment by dantewang** [comment]
ci:test:sf

**Comment by dantewang** [comment]
ci:test:relevant

**Comment by catarineaguiar** [comment]
ci:test:relevant

**Comment by catarineaguiar** [comment]
no regressions found.

File: modules/apps/portal-store/portal-store-gcs/build.gradle [hunk 1]
```diff
@@ -43,4 +43,6 @@ dependencies {
 	compileOnly project(":apps:static:portal-configuration:portal-configuration-metatype-api")
 	compileOnly project(":core:petra:petra-io")
 	compileOnly project(":core:petra:petra-string")
+
+	testCompile group: "com.liferay.portal", name: "com.liferay.util.java", version: "default"
```
Comment (tinatian) [inline modules/apps/portal-store/portal-store-gcs/build.gradle:47]: @brandizzi Why this is needed?

Comment (brandizzi) [reply modules/apps/portal-store/portal-store-gcs/build.gradle]: @tinatian it is not needed anymore. Some previous versions of this PR needed that, but we applied requested changes and this dependency is not necessary anymore. It was removed in this other PR: https://github.com/liferay-core-infra/liferay-portal/pull/2334

**Pull request closed without merging**